jxl/src/features/blending.rs

```diff
@@ -296,13 +323,1855 @@ pub fn perform_blending(
         }
         PatchBlendMode::Mul => {
             for p in 0..3 {
-                perform_mul_blending(bg[p], fg[p], color_blending.clamp, tmp[p]);
+                perform_mul_blending(
+                    bg[p].as_ref(),
+                    fg[p].as_ref(),
+                    color_blending.clamp,
+                    &mut tmp[p],
+                );
             }
         }
         PatchBlendMode::Replace => copy(fg, &mut tmp),
         PatchBlendMode::None => copy(bg, &mut tmp),
     }
     for i in 0..(3 + num_ec) {
-        out[i].copy_from_slice(tmp[i]);
+        out[i].as_mut().copy_from_slice(&tmp[i]);
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    fn clamp(x: f32) -> f32 {
+        x.clamp(0.0, 1.0)
+    }
+
+    const MAX_DELTA: f32 = 1e-6;
+
+    mod perform_alpha_blending_layers_tests {
```

Comment:
Not sure I like the nested test mods...

Reply:
Are you OK enough with them? There are so many tests and so much code that it felt meaningful to separate them a bit, to avoid risking naming collisions and keep things separated.

Reply:
I'll change it later if necessary - submitting now to get on with this :)